src/uct/ib/base/ib_device.c

```diff
@@ -693,18 +687,37 @@ int uct_ib_device_is_port_roce(uct_ib_device_t *dev, uint8_t port_num)
 ucs_status_t uct_ib_device_select_gid_index(uct_ib_device_t *dev,
                                             uint8_t port_num,
                                             size_t md_config_index,
-                                            uint8_t *gid_index)
+                                            uint8_t *gid_index,
+                                            uct_ib_roce_version_desc_t *roce_ver)
```

Comment:
unite both output params to one struct
rename func to uct_ib_device_select_roce_gid_index() and call it only if port is roce